File: src/ros_socket_gqcan_driver/src/subscriber.cpp
#include "ros/ros.h"
#include <can_msgs/gq_car.h>

using namespace std;


void CarMsgsCallback(const can_msgs::gq_car& msg)
{
    ROS_INFO("This is car's current status:");
    ROS_INFO("The car_speed is: %f",msg.car_speed);
    ROS_INFO("The car_acc is: %f",msg.car_acc);
    ROS_INFO("The motor_speed is: %d",msg.motor_speed);
    ROS_INFO("The motor_torque is: %d",msg.motor_torque);
    ROS_INFO("The battery_soc is: %f",msg.battery_soc);
    ROS_INFO("The battery_voltage is: %d",msg.battery_voltage);
    ROS_INFO("The battery_current is: %d",msg.battery_current);
    ROS_INFO("The gas_pedal is: %f",msg.gas_pedal);
    ROS_INFO("The break_pedal is: %f",msg.break_pedal);
    ROS_INFO("The gear_leval is: %f",msg.gear_leval);
    ROS_INFO("The ready_status is: %f",msg.ready_status);

}

int main(int argc,char **argv)
{
    ros::init(argc, argv, "test_listener");

    ros::NodeHandle nh;

    ros::Subscriber sub = nh.subscribe("GQ_Car_Status", 1000, CarMsgsCallback);

    ros::spin();

    return 0;
}

File: src/ros_socket_gqcan_driver/scripts/subscriber.py
#!/usr/bin/env python3
#coding=utf-8

import rospy
from can_msgs import gq_car

def gq_callback(msg):
    rospy.loginfo('The car_speed is: '+str(msg.car_speed))
    rospy.loginfo('The car_acc is: '+str(msg.car_acc))
    rospy.loginfo('The motor_speed is: '+str(msg.motor_speed))
    rospy.loginfo('The motor_torque is: '+str(msg.motor_torque))
    rospy.loginfo('The battery_soc is: '+str(msg.battery_soc))
    rospy.loginfo('The battery_voltage is: '+str(msg.battery_voltage))
    rospy.loginfo('The battery_current is: '+str(msg.battery_current))
    rospy.loginfo('The gas_pedal is: '+str(msg.gas_pedal))
    rospy.loginfo('The break_pedal is: '+str(msg.break_pedal))

if __name__ == '__main__':
    rospy.init_node("test_listener")

    sub = rospy.Subscriber("GQ_Car_Status",gq_car,gq_callback,queue_size = 10)

    rospy.spin()
